Upstream-Status: Backport

- Backported from the ToT: https://crrev.com/c/1213181

Fixes builds errors such as:

error: no match for 'operator=' (operand types are
'{anonymous}::ApplyTextStyleForType(SuggestionAnswer::TextStyle,
OmniboxResultView*, gfx::RenderText*, const
gfx::Range&)::TextStyleNewAnswers' and '<brace-enclosed initializer
list>')
       style = {part_color, .weight = gfx::Font::Weight::BOLD};
                                                             ^
../../chrome/browser/ui/views/omnibox/omnibox_text_view.cc:148:10: note:
candidate: constexpr
{anonymous}::ApplyTextStyleForType(SuggestionAnswer::TextStyle,
OmniboxResultView*, gfx::RenderText*, const
gfx::Range&)::TextStyleNewAnswers&
{anonymous}::ApplyTextStyleForType(SuggestionAnswer::TextStyle,
OmniboxResultView*, gfx::RenderText*, const
gfx::Range&)::TextStyleNewAnswers::operator=(const
{anonymous}::ApplyTextStyleForType(SuggestionAnswer::TextStyle,
OmniboxResultView*, gfx::RenderText*, const
gfx::Range&)::TextStyleNewAnswers&)
   struct TextStyleNewAnswers {
          ^~~~~~~~~~~~~~~~~~~

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From e7a7861f4b0560c618f473a8127bfcd2bae07f2b Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Fri, 7 Sep 2018 18:57:42 +0000
Subject: [PATCH] OmniboxTextView: fix gcc error for structure initialization

It looks like there is bug in GCC 6, which cannot go through
structure initialization normally.

Thus, instead of a default initialization of one of the members,
explicitly initialize it to a default value.

Change-Id: Ia55cc6658e6b6b2f8a80c2582dd28f001c9e648c
Reviewed-on: https://chromium-review.googlesource.com/1213181
Reviewed-by: Scott Violet <sky@chromium.org>
Commit-Queue: Maksim Sisov <msisov@igalia.com>
Cr-Commit-Position: refs/heads/master@{#589614}
---
 chrome/browser/ui/views/omnibox/omnibox_text_view.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/chrome/browser/ui/views/omnibox/omnibox_text_view.cc b/chrome/browser/ui/views/omnibox/omnibox_text_view.cc
index f0a8083dc930..9021284f166d 100644
--- a/chrome/browser/ui/views/omnibox/omnibox_text_view.cc
+++ b/chrome/browser/ui/views/omnibox/omnibox_text_view.cc
@@ -175,7 +175,8 @@ void ApplyTextStyleForType(SuggestionAnswer::TextStyle text_style,
       style = {part_color, .baseline = gfx::SUPERIOR};
       break;
     case SuggestionAnswer::TextStyle::BOLD:
-      style = {part_color, .weight = gfx::Font::Weight::BOLD};
+      style = {part_color, .baseline = gfx::NORMAL_BASELINE,
+               .weight = gfx::Font::Weight::BOLD};
       break;
     case SuggestionAnswer::TextStyle::NORMAL:
     case SuggestionAnswer::TextStyle::NORMAL_DIM:
-- 
2.11.0

